From 79487c9dac7e088b657f7338ed22ddc664c355cf Mon Sep 17 00:00:00 2001
From: Zheng Yejian <zhengyejian1@huawei.com>
Date: Sat, 27 Aug 2022 19:12:42 +0000
Subject: [PATCH 0006/1076] [Huawei] livepatch/core: Fix compile warning when
 print logs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187321

--------------------------------

Fix compile warnings as below:
  include/linux/kern_levels.h:11:18: note: in expansion of macro ‘KERN_SOH’
     11 | #define KERN_ERR KERN_SOH "3" /* error conditions */
        |                  ^~~~~~~~
  include/linux/printk.h:392:9: note: in expansion of macro ‘KERN_ERR’
    392 |  printk(KERN_ERR pr_fmt(fmt), ##__VA_ARGS__)
        |         ^~~~~~~~
  kernel/livepatch/core.c:1045:4: note: in expansion of macro ‘pr_err’
   1045 |    pr_err("%s size less than limit (%lu < %ld)\n",
  func->old_name,
        |    ^~~~~~

Fixes: aba04b9defb0 ("[Huawei] livepatch/core: Restrict minimum size of function that can be patched")
Fixes: af56c7290f29 ("[Huawei] livepatch/x86: Avoid conflict with static {call,key}")
Signed-off-by: Zheng Yejian <zhengyejian1@huawei.com>
---
 kernel/livepatch/core.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/livepatch/core.c b/kernel/livepatch/core.c
index 6e154307159b..daf8d8d618c9 100644
--- a/kernel/livepatch/core.c
+++ b/kernel/livepatch/core.c
@@ -1042,7 +1042,7 @@ static int klp_init_object_loaded(struct klp_patch *patch,
 			return -ENOENT;
 		}
 		if (func->old_size < KLP_MAX_REPLACE_SIZE) {
-			pr_err("%s size less than limit (%lu < %ld)\n", func->old_name,
+			pr_err("%s size less than limit (%lu < %zu)\n", func->old_name,
 			       func->old_size, KLP_MAX_REPLACE_SIZE);
 			return -EINVAL;
 		}
@@ -1217,13 +1217,13 @@ static int check_address_conflict(struct klp_patch *patch)
 			end = start + KLP_MAX_REPLACE_SIZE - 1;
 			ret = jump_label_text_reserved(start, end);
 			if (ret) {
-				pr_err("'%s' has static key in first %ld bytes, ret=%d\n",
+				pr_err("'%s' has static key in first %zu bytes, ret=%d\n",
 				       func->old_name, KLP_MAX_REPLACE_SIZE, ret);
 				return -EINVAL;
 			}
 			ret = static_call_text_reserved(start, end);
 			if (ret) {
-				pr_err("'%s' has static call in first %ld bytes, ret=%d\n",
+				pr_err("'%s' has static call in first %zu bytes, ret=%d\n",
 				       func->old_name, KLP_MAX_REPLACE_SIZE, ret);
 				return -EINVAL;
 			}
-- 
2.17.1

